web/src/selectors/index.js

```diff
@@ -6,44 +6,45 @@ import {
   uniq,
 } from 'lodash';
 
-export function getSelectedZoneHistory(state) {
-  return state.data.histories[state.application.selectedZoneName] || [];
+export function getZoneHistory(zoneId) {
```

Comment:
If all selectors are "accessor" I would add a suffix to the function name to make it clear: `getZoneHistoryAccessor` or `getZoneHistorySelector`. That would make it clear that we're dealing with a higher-order function.
Also it's maybe worth renaming that file to redux.js to make it clear that this file is about redux selectors?

Reply:
Yes, I agree with both of your points - I addressed them in https://github.com/tmrowco/electricitymap-contrib/pull/2317/commits/256eead0c5a372e88d0e83e747caeba24ed929e9 and https://github.com/tmrowco/electricitymap-contrib/pull/2317/commits/00d5965438a74008cc8b046ca00d94159362d8c9.

However, I noticed today that the performance in rendering graphs got much worse after I started using `useSelector` hook and after some research I figured the reason had to do with [shallow compare being replaced by reference comparison](https://react-redux.js.org/api/hooks#equality-comparisons-and-updates), which effectively broke exchange keys calculation memoization and consequently caused `AreaGraph`s to be rerendered continuously when hovering over them.

That and the increased layers of abstraction when using the selectors nudged me to simplify the whole thing further so I actually replaced all those selectors with hooks in https://github.com/tmrowco/electricitymap-contrib/pull/2317/commits/3ff75dbda908ac7d7d9d2e5a3518f75b7a835a05, that do a bit more than just getting the data from Redux - pulling the URL params and doing the memoization is now done from the inside which means the new hooks are much simpler to use in the components, e.g.:

```
const historyData = useCurrentZoneHistory();
```

WDYT?

Reply:
this makes a lot of sense. LGTM